eth2/beacon/types/states.py

```diff
@@ -34,285 +40,270 @@
     ValidatorIndex,
 )
 
-from .blocks import BeaconBlockHeader, BeaconBlock
-from .eth1_data import Eth1Data
-from .eth1_data_vote import Eth1DataVote
-from .crosslinks import Crosslink
-from .forks import Fork
+from .block_headers import (
+    BeaconBlockHeader,
+    default_beacon_block_header,
+)
+from .eth1_data import (
+    Eth1Data,
+    default_eth1_data,
+)
+from .crosslinks import (
+    Crosslink,
+    default_crosslink,
+)
+from .forks import (
+    Fork,
+    default_fork,
+)
 from .pending_attestations import PendingAttestation
 from .validators import Validator
 
+from .defaults import (
+    default_timestamp,
+    default_slot,
+    default_tuple,
+    default_tuple_of_size,
+    default_shard,
+    default_epoch,
+)
+
 
 class BeaconState(ssz.Serializable):
 
     fields = [
-        # Misc
-        ('slot', uint64),
+        # Versioning
         ('genesis_time', uint64),
-        ('fork', Fork),  # For versioning hard forks
+        ('slot', uint64),
+        ('fork', Fork),
+
+        # History
+        ('latest_block_header', BeaconBlockHeader),
+        ('block_roots', Vector(bytes32, 1)),  # Needed to process attestations, older to newer  # noqa: E501
+        ('state_roots', Vector(bytes32, 1)),
+        ('historical_roots', List(bytes32)),  # allow for a log-sized Merkle proof from any block to any historical block root  # noqa: E501
+
+        # Ethereum 1.0 chain
+        ('eth1_data', Eth1Data),
+        ('eth1_data_votes', List(Eth1Data)),
+        ('eth1_deposit_index', uint64),
 
         # Validator registry
-        ('validator_registry', List(Validator)),
-        ('validator_balances', List(uint64)),
-        ('validator_registry_update_epoch', uint64),
-
-        # Randomness and committees
-        ('latest_randao_mixes', Vector(bytes32, 1)),
-        ('previous_shuffling_start_shard', uint64),
-        ('current_shuffling_start_shard', uint64),
-        ('previous_shuffling_epoch', uint64),
-        ('current_shuffling_epoch', uint64),
-        ('previous_shuffling_seed', bytes32),
-        ('current_shuffling_seed', bytes32),
+        ('validators', List(Validator)),
+        ('balances', List(uint64)),
 
-        # Finality
+        # Shuffling
+        ('start_shard', uint64),
+        ('randao_mixes', Vector(bytes32, 1)),
+        ('active_index_roots', Vector(bytes32, 1)),
+
+        # Slashings
+        ('slashed_balances', Vector(uint64, 1)),  # Balances slashed at every withdrawal period  # noqa: E501
```

Comment:
Rename to `slashings`.
```suggestion
        ('slashings', Vector(uint64, 1)),  # Balances slashed at every withdrawal period  # noqa: E501
```